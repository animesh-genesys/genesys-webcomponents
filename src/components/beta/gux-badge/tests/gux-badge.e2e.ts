import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-badge-beta', () => {
  describe('#render', () => {
    [
      '<gux-badge-beta>default</gux-badge-beta>',
      '<gux-badge-beta color="neutral">default explicit</gux-badge-beta>',
      '<gux-badge-beta color="green">sucess</gux-badge-beta>',
      '<gux-badge-beta color="yellow">warning</gux-badge-beta>',
      '<gux-badge-beta color="red">danger</gux-badge-beta>',
      '<gux-badge-beta bold color="green">success bold</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow">warning bold</gux-badge-beta>',
      '<gux-badge-beta bold color="red">danger bold</gux-badge-beta>',
      '<gux-badge-beta color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning icon</gux-badge-beta>',
      '<gux-badge-beta color="red"><gux-icon icon-name="subtract" decorative></gux-icon>danger icon</gux-badge-beta>',
      '<gux-badge-beta color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning icon</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning bold icon</gux-badge-beta>',
      '<gux-badge-beta bold color="red"><gux-icon icon-name="subtract" decorative></gux-icon>danger bold icon</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning bold icon</gux-badge-beta>',
      '<gux-badge-beta bold><gux-icon icon-name="subtract" decorative></gux-icon>default bold</gux-badge-beta>',
      '<gux-badge-beta bold><gux-icon icon-name="subtract" decorative></gux-icon>default bold icon</gux-badge-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-badge-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
