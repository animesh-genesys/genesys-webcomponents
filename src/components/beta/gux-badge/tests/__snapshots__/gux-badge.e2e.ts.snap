// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-badge-beta #render should render component as expected (1) 1`] = `"<gux-badge-beta hydrated=\\"\\">default</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (2) 1`] = `"<gux-badge-beta color=\\"neutral\\" hydrated=\\"\\">default explicit</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (3) 1`] = `"<gux-badge-beta color=\\"green\\" hydrated=\\"\\">sucess</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (4) 1`] = `"<gux-badge-beta color=\\"yellow\\" hydrated=\\"\\">warning</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (5) 1`] = `"<gux-badge-beta color=\\"red\\" hydrated=\\"\\">danger</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (6) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"green\\" hydrated=\\"\\">success bold</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (7) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"yellow\\" hydrated=\\"\\">warning bold</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (8) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"red\\" hydrated=\\"\\">danger bold</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (9) 1`] = `"<gux-badge-beta color=\\"yellow\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>warning icon</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (10) 1`] = `"<gux-badge-beta color=\\"red\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>danger icon</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (11) 1`] = `"<gux-badge-beta color=\\"yellow\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>warning icon</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (12) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"yellow\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>warning bold icon</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (13) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"red\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>danger bold icon</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (14) 1`] = `"<gux-badge-beta bold=\\"\\" color=\\"yellow\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>warning bold icon</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (15) 1`] = `"<gux-badge-beta bold=\\"\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>default bold</gux-badge-beta>"`;

exports[`gux-badge-beta #render should render component as expected (16) 1`] = `"<gux-badge-beta bold=\\"\\" hydrated=\\"\\"><gux-icon icon-name=\\"subtract\\" decorative=\\"\\" hydrated=\\"\\"></gux-icon>default bold icon</gux-badge-beta>"`;
